src/olympia/discovery/views.py

```diff
@@ -1,18 +1,35 @@
 from django_statsd.clients import statsd
 from rest_framework.mixins import ListModelMixin
 from rest_framework.viewsets import GenericViewSet
+from waffle import switch_is_active
 
 from olympia.addons.models import Addon
 from olympia.discovery.data import discopane_items
 from olympia.discovery.serializers import DiscoverySerializer
+from olympia.discovery.utils import get_recommendations, replace_extensions
 
 
 class DiscoveryViewSet(ListModelMixin, GenericViewSet):
     permission_classes = []
     serializer_class = DiscoverySerializer
 
+    def get_discopane_items(self):
+        if not getattr(self, 'discopane_items', None):
+            telemetry_id = (self.kwargs.get('telemetry-client-id') or
+                            self.request.GET.get('telemetry-client-id'))
+            self.discopane_items = discopane_items
+            if switch_is_active('disco-recommendations') and telemetry_id:
```

Comment:
Maybe explicitly check for telemetry id being `None` and force replacements only to happen if that's the case.

Reply:
well, we don't particularly want it if it's an empty string either - it will just end up with a 404 from the recommendations server.